rollup.config.js

```diff
@@ -5,100 +5,82 @@ import { terser } from 'rollup-plugin-terser'
 
 import pkg from './package.json'
 
-export default [
-  // CommonJS
-  {
-    input: 'src/index.js',
-    output: { file: 'lib/redux.js', format: 'cjs', indent: false },
-    external: [
-      ...Object.keys(pkg.dependencies || {}),
-      ...Object.keys(pkg.peerDependencies || {})
-    ],
-    plugins: [babel()]
-  },
+const peersExternal = Object.keys(pkg.peerDependencies || {})
+const allExternal = peersExternal.concat(Object.keys(pkg.dependencies || {}))
 
-  // ES
-  {
-    input: 'src/index.js',
-    output: { file: 'es/redux.js', format: 'es', indent: false },
-    external: [
-      ...Object.keys(pkg.dependencies || {}),
-      ...Object.keys(pkg.peerDependencies || {})
-    ],
-    plugins: [babel()]
-  },
+const makeExternalPredicate = externalArr => {
+  if (externalArr.length === 0) {
+    return () => false
+  }
+  const pattern = new RegExp(`^(${externalArr.join('|')})($|/)`)
+  return id => pattern.test(id)
+}
 
-  // ES for Browsers
-  {
-    input: 'src/index.js',
-    output: { file: 'es/redux.mjs', format: 'es', indent: false },
-    plugins: [
-      nodeResolve({
-        jsnext: true
-      }),
-      replace({
-        'process.env.NODE_ENV': JSON.stringify('production')
-      }),
-      terser({
-        compress: {
-          pure_getters: true,
-          unsafe: true,
-          unsafe_comps: true,
-          warnings: false
-        }
-      })
-    ]
-  },
+const changeExtension = (filename, extension) => filename.replace(/\..+$/i, extension)
 
-  // UMD Development
-  {
-    input: 'src/index.js',
-    output: {
-      file: 'dist/redux.js',
-      format: 'umd',
-      name: 'Redux',
-      indent: false
-    },
-    plugins: [
-      nodeResolve({
-        jsnext: true
-      }),
-      babel({
-        exclude: 'node_modules/**'
-      }),
-      replace({
-        'process.env.NODE_ENV': JSON.stringify('development')
-      })
-    ]
-  },
+const createConfig = ({
```

Comment:
I don't want to use factory functions here. Please keep it as a simple config file. It's harder to read through and figure out what the actual output is this way. Yes, it's more verbose, but it's also much more approachable this way.